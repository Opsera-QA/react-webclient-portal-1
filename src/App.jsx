import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Security, SecureRoute, ImplicitCallback } from '@okta/okta-react';
import Container from 'react-bootstrap/Container'
import { Provider } from "react-redux";
import store from "./store";
import config from './config';
import Home from './Home';
import Navbar from './Navbar';
import Sidebar from './Sidebar';
import Messages from './components/user/Messages';
import Profile from './components/user/Profile';
import About from './components/about/About';
import Pricing from './components/about/Pricing';
import Solutions from './components/about/Solutions';
import Inventory from './components/inventory/Inventory';
import Signup from './components/user/Signup';
import ApiConnector from './components/api_connector/ApiConnector';
import Pipeline from './components/pipeline/Pipeline'
import Platform from './components/platform/Platform';
import Reports from './components/reports/Reports';
import Update from './components/update/Update';

import AdminTools from './components/admin/AdminTools';
import HealthCheck from './components/admin/HealthCheck';
import DeleteTools from './components/admin/DeleteTools';
import RegisteredUsers from './components/admin/RegisteredUsers';
import ManageSystems from './components/admin/ManageSystems';


class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <Security {...config.oidc}>
            <Navbar />          
            <div className="container-fluid">
            <div className="flex-xl-nowrap row sidebar">
              <div className="col-xl-2 col-md-3 col-12 d-flex flex-column bg-dark pt-4">
                <Sidebar /> 
              </div>
              <div className="col-xl-8 col-md-9 col-12 pt-4">
                <Container text>
                  <Route path="/" exact component={Home} />
                  <Route path="/signup" exact component={Signup} />
                  <Route path="/about" exact component={About} />
                  <Route path="/about/pricing" component={Pricing} />
                  <Route path="/about/solutions" component={Solutions} />
                  <Route path="/implicit/callback" component={ImplicitCallback} />
                  <SecureRoute path="/messages" component={Messages} />
                  <SecureRoute path="/profile" component={Profile} />
                  <SecureRoute path="/inventory" component={Inventory} />
                  <SecureRoute path="/api_connector/:id?" component={ApiConnector} />
                  <SecureRoute path="/pipeline" component={Pipeline} />
                  <SecureRoute path="/platform" component={Platform} />
                  <SecureRoute path="/reports" component={Reports} />
                  <SecureRoute path="/update" component={Update} />

                  <SecureRoute path="/admin" exact component={AdminTools} />
                  <SecureRoute path="/admin/health" component={HealthCheck} />
                  <SecureRoute path="/admin/delete" component={DeleteTools} />
                  <SecureRoute path="/admin/manage_systems" component={ManageSystems} />
                  <SecureRoute path="/admin/registered_users" component={RegisteredUsers} />
                </Container>
              </div>
            </div>
            </div>
          </Security>
        </Router>
      </Provider>
    );
  }
}

export default App;