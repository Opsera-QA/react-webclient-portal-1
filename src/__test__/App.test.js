import React from 'react';
import App from '../App';
import {configure, shallow,render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
configure({adapter: new Adapter()});

jest.mock("@okta/okta-signin-widget", () => {
  return {
    OktaSignIn: () => {
      return '';
    }
  };
});

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs/docco", () => {
  return {};
});

jest.mock("react-syntax-highlighter/dist/esm/languages/hljs/xml", () => {
  return '';
});

jest.mock("react-syntax-highlighter", () => {
  return {
    Light: {
      registerLanguage : () => ''
    }

  };
});

it('renders and verifies the snapshot', () => {
  const wrapper = render(<App/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});