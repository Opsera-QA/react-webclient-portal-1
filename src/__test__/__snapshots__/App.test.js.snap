// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and verifies the snapshot 1`] = `
Array [
  <nav
    class="nav-bar navbar navbar-expand navbar-light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <img
        alt="Opsera Inc."
        class="d-inline-block align-top ml-3"
        height="42"
        src="/img/logos/opsera_logo_white_horizontal_240_42.png"
        width="240"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="ml-auto navbar-nav"
      >
        <button
          class="mr-2 btn btn-warning"
          type="button"
        >
          Sign Up
        </button>
        <button
          class="btn btn-outline-success"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </nav>,
  <div />,
  <div
    class="container-fluid"
    style="margin:0"
  >
    <div
      class="d-flex flex-row"
    >
      <div
        class="w-100 pb-4"
      >
        <div
          class="mt-4 row"
        >
          <div
            class="p-4 col-md-5"
          >
            <div
              class="ml-4"
            >
              <h2
                class="mb-3 bd-text-purple-bright"
              >
                Welcome to Opsera!
              </h2>
              <div
                style="font-size:1.1rem"
              >
                Opsera’s vision is to enable and empower the developers, operations and release teams by giving the flexibility in selecting the various DevOps functional tools, build the pipeline with quality and security gates.
              </div>
              <div
                class="mt-3"
                style="font-size:1.1rem"
              >
                Opsera provides out of the box monitoring dashboard, giving an end to end visibility of DevOps landscape metrics via an intelligent dashboard to improve the Agility, Operational excellence and help them to track security and compliance metrics.
              </div>
              <div
                class="row mx-n2 mt-4"
              >
                <div
                  class="col-md px-2"
                >
                  <button
                    class="btn-lg w-100 mb-3 btn btn-primary"
                    type="button"
                  >
                    Add User To Account
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="d-flex align-items-center justify-content-center"
            >
              <div
                class="auth-box-w"
              >
                <div
                  class="logo-w"
                >
                  <img
                    alt="Opsera"
                    height="126"
                    src="/img/logos/opsera_bird_infinity_171_126.png"
                    width="171"
                  />
                </div>
                <h4
                  class="auth-header"
                >
                  Sign in
                </h4>
                <form>
                  <div
                    class="form-group"
                  >
                    <label
                      for="username"
                    >
                      Email Address
                    </label>
                    <input
                      class="form-control"
                      id="username"
                      type="text"
                      value=""
                    />
                    <div
                      class="pre-icon os-icon os-icon-user-male-circle"
                    />
                  </div>
                  <div
                    class="buttons-w"
                  >
                    <button
                      class="w-100 mb-3 btn btn-warning"
                      disabled=""
                      type="submit"
                    >
                      Next
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row fixed-row-footer-bottom"
    >
      <div
        class="col text-center m-1"
        style="padding:0;margin:0;font-size:.6em"
      >
        <span>
          © 2022 Opsera, Inc. The Continuous Orchestration Platform™
        </span>
      </div>
    </div>
  </div>,
]
`;
